packages/react-instantsearch-dom/src/components/__tests__/SmartSort.tsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import renderer from 'react-test-renderer';
+import SmartSort from '../SmartSort';
+
+const textComponent = ({ isSmartSorted }) => (
+  <div>
+    {isSmartSorted
+      ? 'We are filtering unrelevant results'
+      : 'We are not filtering unrelevant results'}
+  </div>
+);
+
+const buttonTextComponent = ({ isSmartSorted }) => (
+  <div>{isSmartSorted ? 'See all results' : 'See relevant results'}</div>
+);
+
+textComponent.propTypes = {
+  isSmartSorted: PropTypes.bool.isRequired,
+};
+
+buttonTextComponent.propTypes = {
+  isSmartSorted: PropTypes.bool.isRequired,
+};
+
+describe('SmartSort', () => {
+  it("returns null if it's not a virtual replica", () => {
+    const tree = renderer.create(
+      <SmartSort
+        buttonTextComponent={buttonTextComponent}
+        isVirtualReplica={false}
+        isSmartSorted={false}
+        refine={() => {}}
+        textComponent={textComponent}
+      />
+    );
+
+    expect(tree.toJSON()).toBeNull();
+  });
+
+  it('accepts a custom className', () => {
+    const tree = renderer.create(
+      <SmartSort
+        buttonTextComponent={buttonTextComponent}
+        className="MyCustomSmartSort"
+        isVirtualReplica={true}
+        isSmartSorted={false}
+        refine={() => {}}
+        textComponent={textComponent}
+      />
+    );
+
+    expect(tree.toJSON()).toMatchSnapshot();
```

Comment:
assert that the root dom has the extra class instead

Reply:
Isn't it the same? As the `.toJSON()` match the snapshot, it means that the classes are equals, no? That's what the other tests do

Reply:
Yes, but a snapshot like this you need to go into another file to check what it means, and it's often tempting to update a whole snapshot file instead of detailed checking what the issue is